Compiling on OSX 10.14 with GIMMI GUI

I'm trying to compiling MicMac with the new GUI GIMMI on OSX 10.14.
I enabled it in cmake list : 

> OPTION(WITH_GIMMI "Graphical Interface GIMMI" ON)
option(DEPLOY "compile for end-user" ON)

Run the cmake with :
> cmake -DWITH_QT5=1 -DCMAKE_PREFIX_PATH=/usr/local/Cellar/qt/5.6.2/ -DNO_X11=1 ..

But I got the following error :

> Scanning dependencies of target GIMMI
1 warning generated.
1 warning generated.
1 warning generated.
1 warning generated.
[ 96%] Linking CXX executable SaisieQT
[ 96%] Building CXX object src/GIMMI/CMakeFiles/GIMMI.dir/Aperi.cpp.o
[ 96%] Building CXX object src/GIMMI/CMakeFiles/GIMMI.dir/C3dc.cpp.o
[ 96%] Building CXX object src/GIMMI/CMakeFiles/GIMMI.dir/Bascule.cpp.o
[ 97%] Building CXX object src/GIMMI/CMakeFiles/GIMMI.dir/Campari.cpp.o
[ 97%] Building CXX object src/GIMMI/CMakeFiles/GIMMI.dir/Cmd.cpp.o
[ 97%] Building CXX object src/GIMMI/CMakeFiles/GIMMI.dir/CropRPC.cpp.o
/Users/anthony/micmac/src/GIMMI/Aperi.cpp:109:26: error: use of undeclared
      identifier 'get_current_dir_name'
            qDebug() <<  get_current_dir_name();
                         ^
/Users/anthony/micmac/src/GIMMI/Bascule.cpp:159:26: error: use of undeclared
      identifier 'get_current_dir_name'
            qDebug() <<  get_current_dir_name();
                         ^
/Users/anthony/micmac/src/GIMMI/Campari.cpp:371:26: error: use of undeclared
      identifier 'get_current_dir_name'
            qDebug() <<  get_current_dir_name();
                         ^
/Users/anthony/micmac/src/GIMMI/Cmd.cpp:79:26: error: use of undeclared
      identifier 'get_current_dir_name'
            qDebug() <<  get_current_dir_name();
                         ^
/Users/anthony/micmac/src/GIMMI/CropRPC.cpp:172:22: error: use of undeclared
      identifier 'get_current_dir_name'
        qDebug() <<  get_current_dir_name();
                     ^
/Users/anthony/micmac/src/GIMMI/C3dc.cpp:424:26: error: use of undeclared
      identifier 'get_current_dir_name'
            qDebug() <<  get_current_dir_name();
                         ^
1 error generated.
1 error generated.
1 error generated.
make[2]: *** [src/GIMMI/CMakeFiles/GIMMI.dir/Aperi.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [src/GIMMI/CMakeFiles/GIMMI.dir/Bascule.cpp.o] Error 1
1 error generated.
make[2]: *** [src/GIMMI/CMakeFiles/GIMMI.dir/Campari.cpp.o] Error 1
1 error generated.
make[2]: *** [src/GIMMI/CMakeFiles/GIMMI.dir/Cmd.cpp.o] Error 1
1 error generated.
make[2]: *** [src/GIMMI/CMakeFiles/GIMMI.dir/CropRPC.cpp.o] Error 1
make[2]: *** [src/GIMMI/CMakeFiles/GIMMI.dir/C3dc.cpp.o] Error 1
make[1]: *** [src/GIMMI/CMakeFiles/GIMMI.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 97%] Built target SaisieQT
[ 97%] Built target mm3d
make: *** [all] Error 2

I've found a possible solution linked to the non-universality of get_current_dir_name() (as not supported python bindings) from the following sources ; [link1](https://github.com/agra-uni-bremen/metaSMT/issues/15) and [link2](https://bugs.python.org/issue1154351) if it is relevant ?

--- a/src/GIMMI/Aperi.cpp
+++ b/src/GIMMI/Aperi.cpp
@@ -106,7 +106,7 @@ void Aperi::mm3d(){
             this->close();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/Bascule.cpp
+++ b/src/GIMMI/Bascule.cpp
@@ -156,7 +156,7 @@ void Bascule::mm3d(){
             this->close();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/C3dc.cpp
+++ b/src/GIMMI/C3dc.cpp
@@ -421,7 +421,7 @@ void C3dc::mm3d(){
             //       cons.show();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/Campari.cpp
+++ b/src/GIMMI/Campari.cpp
@@ -368,7 +368,7 @@ rPCStd.erase(0,4);
             this->close();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/Cmd.cpp
+++ b/src/GIMMI/Cmd.cpp
@@ -76,7 +76,7 @@ void Cmd::mm3d(){
 
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/Convert2GenBundle.cpp
+++ b/src/GIMMI/Convert2GenBundle.cpp
@@ -264,7 +264,7 @@ void Convert2GenBundle::mm3d(){
         //       cons.show();
         p.setWorkingDirectory(path_s.c_str());
         p.waitForFinished(-1);
-        qDebug() <<  get_current_dir_name();
+        qDebug() <<  QDir::currentPath();
         qDebug() <<  cmd.c_str() ;
         p.start(cmd.c_str());
         p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/CropRPC.cpp
+++ b/src/GIMMI/CropRPC.cpp
@@ -169,7 +169,7 @@ void CropRPC::mm3d(){
         this->close();
         p.setWorkingDirectory(path_s.c_str());
         p.waitForFinished(-1);
-        qDebug() <<  get_current_dir_name();
+        qDebug() <<  QDir::currentPath();
         qDebug() <<  cmd.c_str() ;
         p.start(cmd.c_str());
         p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/MM2DPosSism.cpp
+++ b/src/GIMMI/MM2DPosSism.cpp
@@ -134,7 +134,7 @@ void MM2DPosSism::mm3d(){
             this->close();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/MainWindow.cpp
+++ b/src/GIMMI/MainWindow.cpp
@@ -2605,7 +2605,7 @@ void MainWindow::doSomething2(QListWidgetItem *item)
     qDebug() <<  pathProject ;
     p.setWorkingDirectory(pathProject);
     p.waitForFinished(-1);
-    qDebug() <<  get_current_dir_name();
+    qDebug() <<  QDir::currentPath();
     qDebug()<< item->text();
     
     QString cmd = "mm3d Vino "+ pathProject +"/"+ item->text();

--- a/src/GIMMI/Malt.cpp
+++ b/src/GIMMI/Malt.cpp
@@ -477,7 +477,7 @@ void Malt::mm3d(){
             //       cons.show();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/Meshlab.cpp
+++ b/src/GIMMI/Meshlab.cpp
@@ -102,7 +102,7 @@ pyFile = rPC->toPlainText().toStdString();
 
         p.setWorkingDirectory(path_s.c_str());
         p.waitForFinished(-1);
-        qDebug() <<  get_current_dir_name();
+        qDebug() <<  QDir::currentPath();
         qDebug() <<  cmd.c_str() ;
         p.start(cmd.c_str());
         p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/SaisieMasq.cpp
+++ b/src/GIMMI/SaisieMasq.cpp
@@ -83,7 +83,7 @@ void SaisieMasq::mm3d(){
     if(msgBox2.exec() == QMessageBox::Yes){
         p.setWorkingDirectory(path_s.c_str());
         p.waitForFinished(-1);
-        qDebug() <<  get_current_dir_name();
+        qDebug() <<  QDir::currentPath();
         qDebug() <<  cmd.c_str() ;
         p.start(cmd.c_str());
         p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/SaisiePts.cpp
+++ b/src/GIMMI/SaisiePts.cpp
@@ -224,7 +224,7 @@ QMessageBox msgBox2;
             //       cons.show();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/Tapas.cpp
+++ b/src/GIMMI/Tapas.cpp
@@ -900,7 +900,7 @@ void Tapas::mm3d(){
             this->close();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/Tapioca.cpp
+++ b/src/GIMMI/Tapioca.cpp
@@ -402,7 +402,7 @@ cmd = " mm3d Tapioca " + mode +" \""+ images +"\" "+ resom_str +" "+ resoM_str +
         //       cons.show();
         p.setWorkingDirectory(path_s.c_str());
         p.waitForFinished(-1);
-        qDebug() <<  get_current_dir_name();
+        qDebug() <<  QDir::currentPath();
         qDebug() <<  cmd.c_str() ;
         p.start(cmd.c_str());
         p.setReadChannel(QProcess::StandardOutput);

--- a/src/GIMMI/Tarama.cpp
+++ b/src/GIMMI/Tarama.cpp
@@ -215,7 +215,7 @@ QMessageBox msgBox2;
                 //       cons.show();
             p.setWorkingDirectory(path_s.c_str());
             p.waitForFinished(-1);
-            qDebug() <<  get_current_dir_name();
+            qDebug() <<  QDir::currentPath();
             qDebug() <<  cmd.c_str() ;
             p.start(cmd.c_str());
             p.setReadChannel(QProcess::StandardOutput);

